src/mon/PGMap.h

```diff
@@ -171,13 +208,24 @@ class PGMapDigest {
       return p->second.primary;
   }
 
-  ceph_statfs get_statfs() const {
+  ceph_statfs get_statfs(OSDMap &osdmap, boost::optional<int> data_pool) const {
     ceph_statfs statfs;
-    // these are in KB.
-    statfs.kb = osd_sum.kb;
-    statfs.kb_used = osd_sum.kb_used;
-    statfs.kb_avail = osd_sum.kb_avail;
-    statfs.num_objects = pg_sum.stats.sum.num_objects;
+
+    if (data_pool) {
+      const pool_stat_t &stat = pg_pool_sum.at(*data_pool);
```

Comment:
this should probably fall back if the pool doesn't exist (e.g., if a client statfs request races with a pool deletion)